docs/developer-docs/latest/setup-deployment-guides/configurations.md

```diff
@@ -1575,3 +1575,108 @@ module.exports = () => ({
+
+::: warning 🚧 This API is considered unstable for now.
+<br>
+:::
+
+#### Defining a new condition
+
+Conditions are defined as an array of objects. Each condition object can have 4 possible properties:
+
+- `displayName`: the user name displayed in a friendly format, as showed in the admin panel
+- `name`: the condition name, kebab-cased
+- `plugin`: the plugin name, kebab-cased, if your condition is based on a plugin (e.g `content-manager`)
+- `handler`: a function to handle the user and possibly return a `condition object` (see [using the handler function](#using-the-handler-function))
+
+You declare and register conditions in your [`bootstrap`](/developer-docs/latest/setup-deployment-guides/configurations.md#bootstrap) file located at `./config/functions/bootstrap.js` (see [Adding conditions](#adding-conditions)).
+
+#### Using the handler function
+
+The condition `handler` is a synchronous or asynchronous function that:
+
+* receives the authenticated user making the request,
+* and returns `true`, `false`, or another condition object.
+
+Returning `true` means the condition will always match:
+
+```js
+  const condition = {
+    displayName: "Can drink",
+    name: "can-drink",
+    async handler(user) {
+      if (user.age > 21) return true;
+      return false;
+    },
+  };
+```
+
+Returning `false` means the condition will never match:
+
+  ```js
+  const condition = {
+    displayName: "Can drink",
+    name: "can-drink",
+    async handler(user) {
+      if (user.age < 21) return false;
+      return true;
+    },
+  };
+  ```
+
+Returning `true` or `false` is useful if you want to verify an external condition or a condition on the authenticated user only. For more granular control, you can return a condition object.
+
+A condition object is a matcher object used to verify conditions on the entities you `read`, `create`, `update` or `delete`.
+For instance, this custom condition uses an async handler that returns a condition object using the greater than (`$gt`) operator:
+
+```js
+  const condition = {
+    displayName: "price greater than 50",
+    name: "pricate-gt-50",
+    async handler(user) {
+      return { price: { $gt: 50 } };
+    },
+ };
+```
+
+We use [sift.js](https://github.com/crcn/sift.js) behind the scenes to match conditions. Here is the list of allowed operators:
+
+- `$or`
+- `$eq`
+- `$ne`
+- `$in`
+- `$nin`
+- `$lt`
+- `$lte`
+- `$gt`
+- `$gte`
+- `$exists`
+- `$elemMatch`
+
+#### Adding conditions
+
+In your [`./config/functions/bootstrap.js`](/developer-docs/latest.setup-deployment-guides/configurations.md#bootstrap) file, you need to declare conditions and register them, so they are available in the admin panel:
```

Comment:
We should expand on this stating that you construct the conditions but they can be applied dynamically at any time in any environment since the assignment of them is stored in the database, not the file system.

Reply:
Thanks Derrick. I was not aware of this and honestly I'm not sure I fully understand what this implies. So any help of @alexandrebodin would be greatly appreciated on the topic :-)